### PR TITLE
TMT: allow artifact fetching

### PR DESCRIPTION
**Pull request by lsm5**
Testing farm environments set the `TMT_TEST_DATA` variable.
    
From the TMT docs:
TMT_TEST_DATA: Path to the directory where test can store logs and other artifacts generated during its execution.    These will be pulled back from the guest and available for inspection after the test execution is finished.
    
Ref: https://tmt.readthedocs.io/en/stable/overview.html#variables

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -19,7 +19,8 @@ echo "Saving image from image store to filesystem"
 podman save --format oci-archive -o "${OUTDIR}/${DISK_IMAGE_NAME}" "${FULL_IMAGE_NAME_ARCH}"
 
 echo "Transforming OCI image into disk image"
-cd $OUTDIR && sudo sh  ../build-podman-machine-os-disks/build-podman-machine-os-disks.sh "${PWD}/${DISK_IMAGE_NAME}"
+SRCDIR="${TMT_TREE:-..}"
```
Comment (lsm5) [inline build.sh:22]: @baude please make sure this doesn't break on local env for you.

**Comment by baude** [comment]
@lsm5 I will give your PR a whirl and make sure it does not break things.  Also per our discussion, could you consider:

1. adding aarch64
2. renaming podman-machine to something like machine-os-version-arch ?
3. and maybe on #2, add .tar extension
4.

**Comment by lsm5** [comment]
> @lsm5 I will give your PR a whirl and make sure it does not break things. Also per our discussion, could you consider:
> 
> 1. adding aarch64
> 2. renaming podman-machine to something like machine-os-version-arch ?
> 3. and maybe on [Make changes for new repo #2](https://github.com/containers/podman-machine-os/pull/2), add .tar extension

@baude Alright, I guess all done in the latest run:
aarch64: https://artifacts.dev.testing-farm.io/53da412c-b0d5-41ee-a1e6-db87170e55c5/work-build_and_verifyy5nfg5kh/plans/build_and_verify/execute/data/guest/default-0/script-00-1/data/

x86_64: https://artifacts.dev.testing-farm.io/1bc1fc3c-3050-4ecc-b1a9-7839bcb5991b/work-build_and_verifyns6ek5_v/plans/build_and_verify/execute/data/guest/default-0/script-00-1/data/

Also, TMT does allow multi-host testing https://tmt.readthedocs.io/en/stable/spec/plans.html#spec-plans-provision-multihost , so it should be possible to move all artifacts to a single place, but I've yet to experiment with that.

**Comment by lsm5** [comment]
err, yet to rename to `machine-os-`. I'll do that now. Everything else should be good to go.

File: build.sh (hunk 1 above)
Comment (baude) [reply build.sh:22]: tested and verified images built normally and ran

**Comment by lsm5** [comment]
aarch64: https://artifacts.dev.testing-farm.io/1e2bc60b-72db-4067-92d3-1e021a365cc5/work-build_and_verifyhv2_io9y/plans/build_and_verify/execute/data/guest/default-0/script-00-1/data/

x86_64: https://artifacts.dev.testing-farm.io/4251d0b2-a134-4e9e-a228-c162d2fc6459/work-build_and_verify2qqnpx8m/plans/build_and_verify/execute/data/guest/default-0/script-00-1/data/

**Comment by baude** [comment]
yup! LGTM

**Pull request merged**